tests/test_url_handler.py

```diff
@@ -566,7 +567,7 @@ def test_groups_by_prefix_when_many_urls_exist(self):
         for result in json.loads(results):
             assert result in expected
 
-        assert json.loads(results).sort() == expected.sort()
+        # assert json.loads(results).sort() == expected.sort()
```

Comment:
I think this assertion is not necessary 'cos sort() function doesn't return anything, so we are just comparing `None` here.

Reply:
It looks like you found a test that had a bug in it then, but we should try to fix the intent of the test to compare ordered results with ordered what is expected.

Also, based on your comment I notice there is a similar assertion comparing `None`s in `test_returns_expected`. Perhaps the comparison should be using `sorted` function instead of `sort` method.

Reply:
I think, in `create_json_browse`, when urls are >100, we append category to category_list for every url only if already doesn't exist and then returning `category_dict` for those categories in `category_list`. But in the test, we are creating the expected category_dict for every url created by the `SURTFactory` which I think creates duplicate urls. So the lengths of `result` and `expected` are not same. So I removed this check.

Reply:
Is that how the test should be? Or is `expected `supposed to have been set up to match `results`?

Reply:
I checked in python2 in `master` branch. The lengths of `expected` and `results` is not same and the assertion when I used `sorted()` instead of `sort()` failed. I think that is the reason why in the above line `in` is used to check as `assert result in expected`

Reply:
Since this test is meant to be testing the part of `create_json_browse` that produces results without duplicates when the number of urls is greater than 100, then I think that we should be fixing how `expected` has duplicates and is not the same number of items as results. This isn't beautiful, but what about:

```
    def test_groups_by_prefix_when_many_urls_exist(self):
        project = factories.ProjectFactory()
        urls = factories.SURTFactory.create_batch(101, url_project=project)
        root = 'com,'
        results = url_handler.create_json_browse(project.project_slug, None, root)

        # Expected results should be a list without duplicate dict items.
        expected = []
        for url in urls:
            surt_dict = {
                'text': url.value[url.value.find(root) + 4],
                'id': root + url.value[url.value.find(root) + 4],
                'hasChildren': True
            }
            if surt_dict not in expected:
                expected.append(surt_dict)

        sorted_results = sorted(json.loads(results), key=lambda x: x['id'])
        sorted_expected = sorted(expected, key=lambda x: x['id'])
        assert sorted_results == sorted_expected
```

Reply:
While you are fixing buggy tests...could you also fix the other tests that are doing comparisons involving 'sort()` method:
`test_returns_expected`
`test_returns_expected_with_multiple_attribute_values`